tests/python/unittest/test_subgraph_op.py

```diff
@@ -18,351 +18,448 @@
+
+    # infer shape/type before partition before bind
+    check_call(_LIB.MXSetSubgraphPropertyOpNamesV2(c_str(subgraph_backend), mx_uint(len(op_names)),
+                                                 c_str_array(op_names)))
+    part_sym = sym.optimize_for(subgraph_backend, arg_array)
+    check_call(_LIB.MXRemoveSubgraphPropertyOpNamesV2(c_str(subgraph_backend)))
+
+    exe2 = part_sym.bind(ctx=mx.current_context(), args=arg_array, aux_states=aux_array, grad_req='null')
+    exe2.forward()
+    
+    # compare outputs
+    outputs1 = exe1.outputs
+    outputs2 = exe2.outputs
+    assert len(outputs1) == len(outputs2)
+    for i in range(len(outputs1)):
+        assert_almost_equal((outputs1[i] - outputs2[i]).abs().sum().asnumpy(), np.zeros(shape=(1,)))
+    
+def check_subgraph_exe9(sym, subgraph_backend, op_names):
+    inputs = [mx.sym.var(i, dtype=mx_real_t) for i in sym[1]]
+    sym_block = nn.SymbolBlock(sym[0], inputs)
+    sym_block.initialize()
+    shapes = sym[2]
+    x = [mx.nd.random.uniform(shape=s) for s in shapes]
+    outputs1 = sym_block(*x)
+    
+    check_call(_LIB.MXSetSubgraphPropertyOpNamesV2(c_str(subgraph_backend), mx_uint(len(op_names)),
+                                                c_str_array(op_names)))
+    sym_block.hybridize(backend=subgraph_backend)
+    outputs2 = sym_block(*x)
+    check_call(_LIB.MXRemoveSubgraphPropertyOpNamesV2(c_str(subgraph_backend)))
+
+    # compare outputs
+    assert len(outputs1) == len(outputs2)
+    for i in range(len(outputs1)):
+        assert_almost_equal((outputs1[i] - outputs2[i]).abs().sum().asnumpy(), np.zeros(shape=(1,)))
+    
+def check_subgraph(subgraph_backend):
+    for sym, op_names in get_graphs():
+        check_subgraph_exe1(sym[0], subgraph_backend, op_names)
+        check_subgraph_exe2(sym[0], subgraph_backend, op_names)
+        check_subgraph_exe3(sym[0], subgraph_backend, op_names)
+        check_subgraph_exe4(sym[0], subgraph_backend, op_names)
+
+def check_subgraph_backend_sym(subgraph_backend):
+    for sym, op_names in get_graphs():
+        check_subgraph_exe5(sym[0], subgraph_backend, op_names)
+        check_subgraph_exe6(sym[0], subgraph_backend, op_names)
+        check_subgraph_exe7(sym[0], subgraph_backend, op_names)
+        check_subgraph_exe8(sym[0], subgraph_backend, op_names)
+
+def check_subgraph_backend_gluon(subgraph_backend):
+    for sym, op_names in get_graphs():
+        check_subgraph_exe9(sym, subgraph_backend, op_names)
+
+def test_subgraph():
+    check_subgraph('default')
+
+def test_subgraph_v2():
+    check_subgraph('default_v2')
+
+def test_subgraph_backend_sym():
+    check_subgraph_backend_sym('default')
+
+def test_subgraph_backend_sym_v2():
+    check_subgraph_backend_sym('default_v2')
+
+def test_subgraph_backend_gluon():
+    check_subgraph_backend_gluon('default')
+
+def test_subgraph_backend_gluon_v2():
+    check_subgraph_backend_gluon('default_v2')
+
+def test_subgraph_backend_gluon_ext1():
+    def get_net():
```

Comment:
if we dont call get_net more than once lets move the function body out

Reply:
Updated.